src/ostree.cc

```diff
@@ -23,6 +23,32 @@ OstreeSysroot *Ostree::LoadSysroot(const std::string &path) {
   return sysroot;
 }
 
+OstreeDeployment *Ostree::getStagedDeployment(const std::string &path, const std::string &ostree_os) {
+  OstreeSysroot *sysroot = Ostree::LoadSysroot(path);
+  GPtrArray *deployments = NULL;
+  OstreeDeployment *res = NULL;
+
+  const char *osname;
+  if (ostree_os.empty()) {
+    OstreeDeployment *merge_deployment = ostree_sysroot_get_merge_deployment(sysroot, NULL);
+    osname = ostree_deployment_get_osname(merge_deployment);
+  } else {
+    osname = ostree_os.c_str();
+  }
+
+  deployments = ostree_sysroot_get_deployments(sysroot);
+
+  for (unsigned int i = 0; i < deployments->len; i++) {
```

Comment:
Yes, it may make sense. Here we filter out all deployments that don't belong to our OS (taking "OS" as an OSTree term), but since we're only using only one "OS" it only makes things more complicated.